go/private/rules/cgo.bzl

```diff
@@ -31,6 +31,13 @@ load(
     "cc_library",
 )
 
+_WHOLE_ARCHIVE_FLAGS_BY_GOOS = {
+    "darwin": ("-Wl,-force_load", ""),
+    "js": ("", ""),
+    "linux": ("-Wl,--whole-archive", "-Wl,--no-whole-archive"),
+    "windows": ("-Wl,--whole-archive", "-Wl,--no-whole-archive"),
```

Comment:
> Why are these flags OS-specific? I think the assumption here is that clang is always used on darwin, and gcc is always used on linux and windows, but that's not a safe assumption.
> [...]
> Why is "js" included here by the way? What toolchain is expected there?

I cribbed that from https://github.com/bazelbuild/rules_go/pull/1879.

> Is there any way you can get the right flags out of `cc_common` from the toolchain? If not, I think it would be better to look at the name of the compiler or linker binary to guess which flags will be accepted.

If there is a way to do that, that would be great; I have no idea, though. There doesn't appear to be any sort of sort of sanctioned forum to ask the devs and others in the know, and I have no ties with Google or the Bazel devs, so I don't know how to inquire. Any advice here would be great.

Reply:
I don't see anything directly in `cc_common` that would help.

I think a "good enough" solution would be to check `go.env["CC"]`, which will be the `CC` environment variable passed to actions. If the string after the last slash contains `clang`, let's use the clang flags. Otherwise, let's use the gcc flags. I'm not 100% sure that will always be set, so let's assume gcc if it's not.

(For asking questions by the way, the [bazel-discuss](https://groups.google.com/g/bazel-discuss) mailing list is usually pretty good. There's also a public Slack, though I'm not sure if it's still active).